ragthoven/tools/example_tool.py

```diff
@@ -0,0 +1,28 @@
+"""
+Every tool is a function that takes the data row in form of dictionary in a variable `args` and can modify it.
+- Every tool must return the said args in the end.
+- The tools are called sequentially as specified in yaml config.
+"""
+
+
+def fizzbuzz(args):
+    text = str(args["article"])
+    n_of_spaces = text.count(" ")
+    args["fizzbuzz"] = (
+        "fizzbuzz"
+        if n_of_spaces % 3 and n_of_spaces % 5
+        else "buzz" if n_of_spaces % 3 else "fizz" if n_of_spaces % 5 else "Nothing"
+    )
+    return args
+
+
+def count_ands(args):
+    text = str(args["article"])
+    args["and_countes"] = text.count("and")
+    return args
+
+
+def replace_text_with_length(args):
```

Comment:
@gkaretka what if you wanted to use a key different from `article` but the same function? Or is it too soon to think about that?

Reply:
I think that the python code provides a good enough flexibility but also, it might make sense to implement some kind of `in`s and `out`s later on in `yaml` for even better configurability.